### PR TITLE
Refactor : replace Proxy Container in scaling with the one in env module(#19956)

### PR DESCRIPTION
**Pull request by taojintianxia**

### CODE REVIEW

File: shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-scaling/src/test/java/org/apache/shardingsphere/integration/data/pipeline/framework/container/compose/DockerComposedContainer.java [hunk 1]
```diff
@@ -33,7 +34,7 @@ public final class DockerComposedContainer extends BaseComposedContainer {
     
     private final DatabaseType databaseType;
     
-    private final ShardingSphereProxyDockerContainer proxyContainer;
+    private final ShardingSphereProxyClusterContainer proxyContainer;
```
Comment (TeslaCN) [inline shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-scaling/src/test/java/org/apache/shardingsphere/integration/data/pipeline/framework/container/compose/DockerComposedContainer.java:37]: Could we remove `ShardingSphereProxyDockerContainer` in this PR?

Comment (taojintianxia) [reply shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-scaling/src/test/java/org/apache/shardingsphere/integration/data/pipeline/framework/container/compose/DockerComposedContainer.java:37]: sure thing. I intended to remove it in next PR :)

File: shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-scaling/src/test/resources/env/mysql/server-5.yaml [hunk 2]
```diff
@@ -31,7 +31,7 @@ mode:
 rules:
   - !AUTHORITY
     users:
-      - root@:Root@123
+      - proxy@:Proxy@123
```
Comment (TeslaCN) [inline shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-scaling/src/test/resources/env/mysql/server-5.yaml:34]: Are these files under shardingsphere-integration-test-scaling still used? You have copied them to test-env.

File: shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-scaling/src/test/java/org/apache/shardingsphere/integration/data/pipeline/framework/container/compose/DockerComposedContainer.java (hunk 1 above)
Comment (TeslaCN) [reply shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-scaling/src/test/java/org/apache/shardingsphere/integration/data/pipeline/framework/container/compose/DockerComposedContainer.java:37]: > sure thing. I intended to remove it in next PR :)

It's better to remove unused files or codes in time.

File: shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-scaling/src/test/resources/env/mysql/server-5.yaml (hunk 2 above)
Comment (taojintianxia) [reply shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-scaling/src/test/resources/env/mysql/server-5.yaml:34]: there are some other case still need these config files. I don't want to make this PR too huge .

**Review by terrymanu — APPROVED** [review]

File: shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-scaling/src/test/java/org/apache/shardingsphere/integration/data/pipeline/framework/container/compose/DockerComposedContainer.java (hunk 1 above)
Comment (taojintianxia) [reply shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-scaling/src/test/java/org/apache/shardingsphere/integration/data/pipeline/framework/container/compose/DockerComposedContainer.java:37]: > > sure thing. I intended to remove it in next PR :)
> 
> It's better to remove unused files or codes in time.

yeah, you are right.

**Pull request merged**